pkg/daemon/manager/manager.go

```diff
@@ -19,6 +25,9 @@ type Manager interface {
 	DeleteService(string) error
 	// GetExistingServices is a wrapper of ListServices, returning a map of all services that exist on an instance
 	GetExistingServices() (map[string]struct{}, error)
+	// EnsureServiceIsRemoved deletes the given service and all services that depend on it, even those unmanaged by WICD
+	// If the managedOnly flag is true, only services with the managed tag will be removed.
+	EnsureServiceIsRemoved(string, bool) error
```

Comment:
I'm not sold on both `DeleteService` and this method being necessary.
Do you have any reason for them to not be combined?

Reply:
I wanted to keep this separate as it also deletes dependencies. It felt right to provide a less expensive option by exposing the `DeleteService` method instead of just making it a helper

Reply:
Thanks that reasoning makes sense.

In that case two methods can make sense. But I see no reason for `DeleteService` to not 'ensure' that the service is deleted, and not throw an error if it doesnt exist.

I don't see a usecase for WICD where failure upon the service not existing is useful. Exposing the real API call's behavior isn't helpful for us, and our code can be simplified if both interface methods maintain the same behavior in the case that the service does not exist. This change would require a rename of both methods for clarity.

Reply:
I'm thinking through this more, and I think what might be complicating things here is adding the concept of a managed service to the `Manager` interface. Bringing that concept out of this package is something that should be considered.

Additionally, why does the logic around the manged services need to exist? What services not managed by WICD are going to depend on services managed by WICD? I don't see this as a case we should be willing to support as it will prevent WICD from performing its required functionality.

Reply:
> What services not managed by WICD are going to depend on services managed by WICD?

One use case I can think of -- it's needed for the intermediate time period where some services are WICD managed and some are not. For example an unmanaged hybrid-overlay may depend on a WICD-managed kubelet service.